src/components/Sidebar/Sidebar.jsx

```diff
@@ -0,0 +1,152 @@
+    }
+
+    const cancel = () => {
+        recentlySelected.map(({ id, isCategory }) => {
+            isCategory
+                ? dispatch(setCategory(null))
+                : dispatch(toggleProvider(id))
+        })
+
+        changeFiltersVisibility()
+        setRecentlySelected([])
+    }
+
+    const applyFilters = () => {
+        changeFiltersVisibility()
+        setRecentlySelected([])
+    }
+
+    const renderAllCategories = () =>
+        categories.map(category => (
+            <div
+                key={category.id}
+                className={`${styles['category']} ${
+                    category.id === selectedCategory ? styles['selected'] : ''
+                }`}
+                onClick={select.bind(this, category.id, true)}
+            >
+                <img src={category.icon} alt="category-icon" />
+                <p>{category.displayName}</p>
+            </div>
+        ))
+
+    const renderAllProviders = () =>
+        providers.map(provider => (
+            <div
+                key={provider.id}
+                className={`${styles['provider']} ${
+                    selectedProviders.includes(provider.id)
+                        ? styles['selected']
+                        : ''
+                }`}
+                onClick={select.bind(this, provider.id, false)}
+            >
+                <img src={provider.icon} alt="provider-icon" />
+            </div>
+        ))
+
+    const handleFilterButton = () => {
+        return !filtersVisibility ? (
+            <button
+                className={styles['filters-closed']}
+                onClick={changeFiltersVisibility}
+            >
+                <img src="./icons/settings.svg" alt="filters-icon" />
+                <span>Фильтры</span>
+            </button>
+        ) : (
+            <div className={styles['filters-opened']}>
+                <div>
+                    <span>Фильтры</span>
+                    <img
+                        src="./icons/cancel.svg"
+                        onClick={cancel}
+                        alt="cancel-icon"
+                    />
+                </div>
+
+                <hr />
+            </div>
+        )
+    }
+
+    const renderApplyFilterButton = () =>
+        width === MOBILE_WIDTH &&
+        (selectedCategory || selectedProviders.length !== 0) && (
+            <button
+                className={styles['apply-filters-button']}
+                onClick={applyFilters}
+            >
+                Применить
+            </button>
+        )
+
+    const renderOpenedFilters = () => {
+        return (
+            <>
+                <div className={styles['categories']}>
+                    {renderAllCategories()}
+                </div>
+
+                <hr />
+
+                <div className={styles['providers']}>
+                    <h2>Провайдеры</h2>
+                    {renderAllProviders()}
+                </div>
+
+                {renderApplyFilterButton()}
```

Comment:
we can just create a button here and hide/show it depending on the settings rather than re-generate it every time, IMHO

Reply:
Discussed this with Andrii Halich, he said it wouldn`t harm the productivity and the speed of an app and suggested to move some logic to different components, so created a separate component for Categories, Providers and Filter button

Reply:
got it 👍